nomad/acl.go

```diff
@@ -118,7 +118,7 @@ func (s *Server) Authenticate(ctx *RPCContext, secretID string) (*structs.Authen
 }
 
 func (s *Server) ResolveACL(aclToken *structs.ACLToken) (*acl.ACL, error) {
-	if !s.config.ACLEnabled {
+	if !s.config.ACLEnabled || aclToken == nil {
```

Comment:
This line is a good candidate for a helper func as well as (1) I'd be surprised if accessing `s.config` *wasn't* a race condition and therefore needs locking added someday, and (2) really easy to get a single character wrong in this line and cause a hard to spot security issue.

It's a well trod path though, so not a blocker this work.

Reply:
I was just in `Server.Reload` and it looks like `s.config` isn't swapped and `s.config.ACLEnabled` isn't touched, so this probably isn't a race condition (1)

Reply:
This is actually just totally wrong, so thanks for catching it. I made this change to fix a NPE coming from variables (which may have claims instead of an ACL token), but that's not safe to do for the other consumers of this function. I'll back this bit out and check for the nil pointer in the caller in variables instead.